### ISSUE TITLE
KERNEL_SECURITY_CHECK_FAILURE with device=GPU on Windows

### ISSUE THREAD

**Issue by sergeyf** [Theano/Theano#3820]
I'm on Windows 8.1, using the latest Theano. Not 100% sure when this started, but now when `device = gpu` in .theanorc, doing `import theano` causes a blue screen of death with a KERNEL_SECURITY_CHECK_FAILURE, followed by a forced restart. 

I've tried deleting everything in the compile directory - doesn't help. Here's my complete `.theanorc`:

```
[global]
device = gpu
force_device = True
floatX = float32
mode = FAST_RUN
 
[nvcc]
compiler_bindir = D:\VS12\VC\bin\amd64
LC=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.5\lib\x64
cuda.root=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.5\bin
```

**Comment by sergeyf** [comment]
Another observation. I did a fresh `python setup.py install`, set `force_device = False`, and redeleted everything in the compile directory. trying `import theano`, I now get:

```
['nvcc', '-shared', '-O3', '--compiler-bindir', 'D:\\VS12\\VC\\bin\\amd64', '-Xlinker', '/DEBUG', '-D HAVE_ROUND', '-m64', '-Xcompiler', '-DCUDA_NDARR
AY_CUH=18715462c72ed6afcd7ca5d52813ce90,-DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION,/Zi,/MD', '-IC:\\Users\\S\\Documents\\GitHub\\theano\\theano\\sand
box\\cuda', '-ID:\\Anaconda2\\lib\\site-packages\\numpy\\core\\include', '-ID:\\Anaconda2\\include', '-IC:\\Users\\S\\Documents\\GitHub\\theano\\thean
o\\gof', '-o', 'C:\\Users\\S\\AppData\\Local\\Theano\\compiledir_Windows-8.1-6.3.9600-Intel64_Family_6_Model_60_Stepping_3_GenuineIntel-2.7.11-64\\cud
a_ndarray\\cuda_ndarray.pyd', 'mod.cu', '-LD:\\Anaconda2\\libs', '-LD:\\Anaconda2', '-lpython27', '-lcublas', '-lcudart']
ERROR (theano.sandbox.cuda): Failed to compile cuda_ndarray.cu: ('nvcc return status', 2, 'for cmd', 'nvcc -shared -O3 --compiler-bindir D:\\VS12\\VC\
\bin\\amd64 -Xlinker /DEBUG -D HAVE_ROUND -m64 -Xcompiler -DCUDA_NDARRAY_CUH=18715462c72ed6afcd7ca5d52813ce90,-DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERS
ION,/Zi,/MD -IC:\\Users\\S\\Documents\\GitHub\\theano\\theano\\sandbox\\cuda -ID:\\Anaconda2\\lib\\site-packages\\numpy\\core\\include -ID:\\Anaconda2
\\include -IC:\\Users\\S\\Documents\\GitHub\\theano\\theano\\gof -o C:\\Users\\S\\AppData\\Local\\Theano\\compiledir_Windows-8.1-6.3.9600-Intel64_Fami
ly_6_Model_60_Stepping_3_GenuineIntel-2.7.11-64\\cuda_ndarray\\cuda_ndarray.pyd mod.cu -LD:\\Anaconda2\\libs -LD:\\Anaconda2 -lpython27 -lcublas -lcud
art')
WARNING (theano.sandbox.cuda): CUDA is installed, but device gpu is not available  (error: cuda unavailable)
```

**Comment by nouiz** [comment]
It seem that now your GPU isn't well configured in the OS. Try to compile
and run basic cuda example and see if they work. They should also fail.

Do you have a good driver? Do you need to reboot?

**Comment by sergeyf** [comment]
Howdy,

I have restarted many times, and reinstalled the latest drivers from Nvidia's site.

I opened up Samples_vs2013.sln in Visual Studio and started building random sample projects, and they all seem to compile without issue.

However, after I compiled deviceQuery.exe and tried to execute it - I got the same BSOD. So it is something with my CUDA config. 

Thanks for the tip! I'll leave this ticket open for now so I can report what the fix is (if I ever find it).

**Comment by eyaler** [comment]
seeing the same issue

**Comment by nouiz** [comment]
You should check on nvidia forum. They will be more able to help on this.

**Comment by sergeyf** [comment]
Here's the topic: https://devtalk.nvidia.com/default/topic/910190/cuda-setup-and-installation/kernel_security_check_failure-when-running-devicequery-exe-on-windows-8-1/

**Comment by sergeyf** [comment]
I fixed it by downgrading to driver version 353.62 and checking the Clean Install box when doing the Custom Installation (not Express) route.

The suggestion came from here: http://answers.microsoft.com/en-us/windows/forum/windows_10-performance/bsod-kernel-security-check-failure-when-opening/a5b8c833-2118-4f28-867e-09820af7779d?auth=1

Thanks for the suggestion of posting on the nvidia forum! 

I very much appreciate all the hard work that happens on Theano - it is a great package.
